pkg/config/config_template.yaml

```
@@ -413,6 +413,136 @@ api_key:
 #
 # secret_backend_timeout: 5
 
+## @param snmp_listener - custom object - optional
+## Creates and schedules a listener to automatically discover your SNMP devices.
+## Discovered devices can then be monitored with the SNMP integration by using
+## the auto_conf.yaml file provided by default.
+#
+# snmp_listener:
+
+  ## @param workers - integer - optional - default: 2
+  ## The number of concurrent tasks used to discover SNMP devices. Increasing this value
+  ## discovers devices faster but at the cost of increased resource consumption.
+  #
+  # workers: 2
+
+  ## @param discovery_interval - integer - optional - default: 3600
+  ## How often to discover new SNMP devices, in seconds. Decreasing this value
+  ## discovers devices faster (within the limit of the time taken to scan subnets)
+  ## but at the cost of increased resource consumption.
+  #
+  # discovery_interval: 3600
+
+  ## @param allowed_failures - integer - optional - default: 3
+  ## The number of failed requests to a given SNMP device before removing it from the list of monitored
+  ## devices.
+  ## If a device shuts down, the Agent stops monitoring it after `discovery_interval * allowed_failures` seconds.
+  #
+  # allowed_failures: 3
+
+  ## @param configs - list - required
+  ## The actual list of configurations used to discover SNMP devices in various subnets.
+  ## Example:
+  ## configs:
+  ##  - network: 10.0.0.0/24
+  ##    version: 2
+  ##    community: public
+  ##  - network: 10.0.1.0/28
+  ##    ignored_ip_addresses:
+  ##      - 10.0.0.1.0
+  ##      - 10.0.0.1.1
```

New example on top of detailed configuration parameters looks great!

But did you mean…?

```suggestion
  ##      - 10.0.1.0
  ##      - 10.0.1.1
```

Besides - is the second config valid, i.e. isn't it missing a `community` (or `user`)?

---

This should be valid for SNMP v1 right?

---

Changed to explicit v1 and implicit v2